src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc: GroupsMap<T>, item: T) => {
+    const groupKey = String(item[key]);
+
+    if (acc[groupKey]) {
+      acc[groupKey].push(item);
+    } else {
+      acc[groupKey] = [item];
+    }
+
+    return acc;
+  }, {});
```

Comment:
Consider initializing the accumulator with a more specific type signature directly in the `reduce` function. This will make the code more robust by not relying on type inference, which can be especially useful if the `items` array is empty. You can initialize the accumulator as `{} as GroupsMap<T>`.